### PR TITLE
Get current usage for metered billing in store.

### PR DESCRIPTION
**Pull request by chamindias**
This PR introduces the get current usage (metered billing) for monetization.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 1]
```diff
@@ -2527,6 +2527,54 @@ paths:
...
+    get:
+      x-wso2-curl: "curl -k -H \"Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8\" https://127.0.0.1:9443/api/am/store/v1.0/subscriptions/64eca60b-2e55-4c38-8603-e9e6bad7d809/usage"
+      x-wso2-request: |
+        GET https://127.0.0.1:9443/api/am/store/v1.0/subscriptions/64eca60b-2e55-4c38-8603-e9e6bad7d809/usage
+        Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8
+      x-wso2-response: "HTTP/1.1 200 OK\nContent-Type: application/json\n\n{\n   \"object\": \"invoice\",\n   \"account_country\": \"US\",\n   \"account_name\": \"test.com\",\n   \"amount_due\": \"0\",\n   \"amount_paid\": \"0\"\n  \"amount_remaining\": \"0\"\n}"
+      security:
+        - OAuth2Security:
+          - apim:sub_manage
+      summary: Get details of a pending invoice for a monetized subscription with meterd billing.
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]:
```suggestion
      summary: Get details of a pending invoice for a monetized subscription with metered billing.
```

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java [hunk 2]
```diff
@@ -357,6 +362,35 @@ public Response subscriptionsSubscriptionIdGet(String subscriptionId, String ifN
...
+                String errorMessage = "Billing engine usage data was not found for subscription ID : " + subscriptionId;
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+            APIMonetizationUsageDTO apiMonetizationUsageDTO = new APIMonetizationUsageDTO();
+            apiMonetizationUsageDTO.setProperties(billingEngineUsageData);
+            return Response.ok().entity(apiMonetizationUsageDTO).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Failed to retrieve billing engine usage data for subscription ID : " + subscriptionId;
+            RestApiUtil.handleInternalServerError(errorMessage, log);
+        } catch (MonetizationException e) {
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java]: We must use `e` in the log in all error cases.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml (hunk 1 above)
Comment (chamindias) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: Done in : https://github.com/wso2/carbon-apimgt/pull/6961/commits/0832832ce6d6c5e07bb85ed4d0846dafa034f88b

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java (hunk 2 above)
Comment (chamindias) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java]: Done in : https://github.com/wso2/carbon-apimgt/pull/6961/commits/0832832ce6d6c5e07bb85ed4d0846dafa034f88b

Comment (chamindias) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SubscriptionsApiServiceImpl.java]: Done in : https://github.com/wso2/carbon-apimgt/pull/6961/commits/0832832ce6d6c5e07bb85ed4d0846dafa034f88b

**Pull request merged**